### ISSUE TITLE
Ability to support different events for triggering escape outside

### ISSUE THREAD

**Issue by etyp** [amytych/react-escape-outside#3]
Thanks for the package - it's super valuable for us. We're looking to trigger escape on other events aside from `click` - specifically, we have a case where we want to support `mousedown` instead so it triggers immediately.

I've put together a sample gist for how it might work: https://gist.github.com/etyp/0c75e73bfb2efe74eb4cd91c62c54d3a

But basically the idea would be usage looks like passing a `mouseEvent` and/or `touchEvent` prop like so:

```
<EscapeOutside mouseEvent="mousedown" touchEvent="touchstart">
```

Not passing these props defaults to `click` and `touchend`. If we create a PR would you consider merging and publishing?

**Comment by amytych** [comment]
Thanks for the suggestion this seems like a useful addition, I'd be happy to take a PR on that.

**Comment by amytych** [comment]
Closed via #4
